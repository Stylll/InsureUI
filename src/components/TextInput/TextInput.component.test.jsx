import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import TextInput from './TextInput.component';

describe('TextInput Component', () => {
  it('should render properly', () => {
    const { baseElement } = render(<TextInput
        name="name"
        title="Item Name"
        value="Television"
        type="text"
        error=""
        placeholder="enter your item name"
        onChange={() => {}}
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly with error message', () => {
    const { baseElement } = render(<TextInput
        name="name"
        title="Item Name"
        value="Television"
        type="text"
        error="name is required"
        placeholder="enter your item name"
        onChange={() => {}}
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onchange method when value is typed', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<TextInput
        name="name"
        title="Item Name"
        value="Television"
        type="text"
        error="name is required"
        placeholder="enter your item name"
        onChange={onChange}
        dataTestId="textinput"
        />);

    fireEvent.change(getByTestId('textinput'), { target: { value: '123' } });
    expect(onChange).toHaveBeenCalled();
  });
});
