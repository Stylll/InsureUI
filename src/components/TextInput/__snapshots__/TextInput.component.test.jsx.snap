// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should render properly 1`] = `
<body>
  <div>
    <div
      class="form-input"
    >
      <div
        class="title"
      >
        Item Name
      </div>
      <input
        class="text-input"
        data-testid="textinput"
        id=""
        name="name"
        placeholder="enter your item name"
        type="text"
        value="Television"
      />
      <span
        class="info"
      />
      
    </div>
  </div>
</body>
`;

exports[`TextInput Component should render properly with error message 1`] = `
<body>
  <div>
    <div
      class="form-input"
    >
      <div
        class="title"
      >
        Item Name
      </div>
      <input
        class="text-input"
        data-testid="textinput"
        id=""
        name="name"
        placeholder="enter your item name"
        type="text"
        value="Television"
      />
      <span
        class="error"
        id="name-error"
      >
        name is required
      </span>
    </div>
  </div>
</body>
`;
